store/governance/proposal.js

```diff
@@ -0,0 +1,257 @@
+      .divRound(resultAll)
+      .toNumber() / 100
+
+  return {
+    id: `${transactionHash}-${voter}`,
+    percentage,
+    ...returnValues,
+    votes: fromWei(returnValues.votes),
+    transactionHash,
+    blockNumber,
+
+    ...comment,
+
+    ens: {
+      delegator: null,
+      voter: null
+    },
+    delegator: null,
+    timestamp: null
+  }
+}
+
+const createFetchCommentsWithMessages = (web3, batch, govInstance) => async (proposalComment) => {
+  const { transactionHash, voter, blockNumber } = proposalComment
+
+  if (!CACHE_TX[transactionHash]) {
+    CACHE_TX[transactionHash] = new Promise((resolve, reject) => {
+      const callback = createBatchRequestCallback(resolve, reject)
+      batch.add(web3.eth.getTransaction.request(transactionHash, callback))
+    })
+  }
+
+  if (!CACHE_BLOCK[blockNumber]) {
+    CACHE_BLOCK[blockNumber] = new Promise((resolve, reject) => {
+      const callback = createBatchRequestCallback(resolve, reject)
+      batch.add(web3.eth.getBlock.request(blockNumber, callback))
+    })
+  }
+
+  try {
+    const [tx, blockInfo] = await Promise.all([CACHE_TX[transactionHash], CACHE_BLOCK[blockNumber]])
+
+    const isMaybeHasComment = voter === tx.from
+    const comment = parseComment(isMaybeHasComment ? tx.input : null, govInstance)
+
+    return {
+      ...proposalComment,
+      ...comment,
+
+      delegator: voter === tx.from ? null : tx.from,
+      timestamp: blockInfo.timestamp
+    }
+  } catch (error) {
+    CACHE_TX[transactionHash] = null
+    CACHE_BLOCK[blockNumber] = null
+    return proposalComment
+  }
+}
+
+const state = () => {
+  return {
+    isFetchingComments: false,
+    isFetchingMessages: false,
+    ensNames: {},
+    comments: []
+  }
+}
+
+const getters = {
+  comments: (state) => {
+    const { ensNames } = state
+    let comments = state.comments.slice()
+
+    comments.sort((a, b) => b.blockNumber - a.blockNumber)
+    comments = uniqBy(comments, 'voter')
+    comments.sort((a, b) => b.percentage - a.percentage)
+
+    comments = comments.map((data) => ({
+      ...data,
+      ens: {
+        delegator: ensNames[data.delegator],
+        voter: ensNames[data.voter]
+      }
+    }))
+
+    return comments
+  }
+}
+
+const mutations = {
+  SAVE_FETCHING_COMMENTS(state, status) {
+    state.isFetchingComments = status
+  },
+  SAVE_FETCHING_MESSAGES(state, status) {
+    state.isFetchingMessages = status
+  },
+  SAVE_ENS_NAMES(state, ensNames) {
+    state.ensNames = { ...state.ensNames, ...ensNames }
+  },
+  SAVE_COMMENTS(state, comments) {
+    state.comments = comments
+  }
+}
+
+const actions = {
+  async fetchComments(context, proposal) {
+    const { commit, dispatch, state } = context
+    let { comments } = state
+
+    if (comments[0]?.id !== proposal.id) {
+      commit('SAVE_COMMENTS', [])
+      comments = []
+    }
+
+    commit('SAVE_FETCHING_COMMENTS', true)
+    const newComments0 = await dispatch('fetchVotedEvents', { proposal, comments })
```

Comment:
Variable names with numbers should be avoided

Reply:
done